rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -3133,9 +3284,34 @@ extern "C" rmw_ret_t rmw_send_response(
   RET_NULL(ros_response);
   CddsService * info = static_cast<CddsService *>(service->data);
   cdds_request_header_t header;
-  memcpy(&header.guid, request_header->writer_guid, sizeof(header.guid));
+  dds_instance_handle_t reqwrih;
+  static_assert(
+    sizeof(request_header->writer_guid) == sizeof(header.guid) + sizeof(reqwrih),
+    "request header size assumptions not met");
+  memcpy(
+    static_cast<void *>(&header.guid), static_cast<const void *>(request_header->writer_guid),
+    sizeof(header.guid));
+  memcpy(
+    static_cast<void *>(&reqwrih),
+    static_cast<const void *>(request_header->writer_guid + sizeof(header.guid)), sizeof(reqwrih));
   header.seq = request_header->sequence_number;
-  return rmw_send_response_request(&info->service, header, ros_response);
+  // Block until the response reader has been matched by the response writer (this is a
```

Comment:
@eboasson mind to add a TODO for a proper fix? I've opened #191 to track that work.